pkg/arvo/app/chat.hoon

```diff
@@ -340,201 +222,197 @@
-                  configs.str.sta
-                (~(put by configs.str.sta) affectedcir ~)
-            ==
-          ::
-          :_  this(str.sta str)
-          %+  weld
-            [ost.bol %peer newwir [hos.affectedcir %hall] pat]~
-            (send-chat-update [[%inbox newinbox] str])
-        ::
-        =/  newinbox  %=  inbox.str.sta
-          src  (~(del in src.inbox.str.sta) src.dif.sto)
+    ::  %remove: remove a circle
+    ::
+        %remove
+      =.  str
+        %=  str
+          configs   (~(del by configs.str) circ)
+          messages  (~(del by messages.str) circ)
+          peers     (~(del by peers.str) circ)
         ==
-        ::  we've removed a source from our inbox
-        ::
-        =/  str
-          %=  str.sta
-            inbox  newinbox
-          ::
-            configs   (~(del by configs.str.sta) affectedcir)
-            messages  (~(del by messages.str.sta) affectedcir)
-            peers     (~(del by peers.str.sta) affectedcir)
-          ==
-        =/  fakecir/circle:hall
-          :-  our.bol
-          %-  crip
-          %+  weld  (trip 'hall-internal-')  (trip nom.affectedcir)
-        ::
-        ?~  (~(get by configs.str) fakecir)
-          ::  just forward the delete to our clients
-          ::
-          :_  this(str.sta str)
-          %+  weld
-            [ost.bol %pull newwir [hos.affectedcir %hall] ~]~
-          %+  weld
-            (send-chat-update [[%inbox newinbox] str])
-            (send-chat-update [[%delete affectedcir] str])
-        ::  if we get a delete from another ship, delete our fake circle copy
-        ::
-        =/  deletefake/poke
-          :-  %hall-action
-              [%delete nom.fakecir ~]
-          :_  this(str.sta str)
-          %+  weld
-            [ost.bol %pull newwir [hos.affectedcir %hall] ~]~
-          %+  weld
-            [ost.bol %poke /fake [our.bol %hall] deletefake]~
-          %+  weld
-            (send-chat-update [[%inbox newinbox] str])
-            (send-chat-update [[%delete affectedcir] str])
+      :_  this(str str)
+      (send-chat-update [[%delete circ] str])
       ::
-      ::  %remove: remove a circle
-      ::
-          %remove
-        =/  str
-          %=  str.sta
-            configs   (~(del by configs.str.sta) circ)
-            messages  (~(del by messages.str.sta) circ)
-            peers     (~(del by peers.str.sta) circ)
-          ==
-        :-  (send-chat-update [[%delete circ] str])
-        this(str.sta str)
-        ::
-      ==
-      ::  end of branching on dif.sto type
     ==
-    ::  end of branching on sto type
   ==
-  ::  end of i.wir branching
 ::
-::  +lient arms
+++  handle-rumor-circle-gram
+  |=  [wir=wire sto=rumor-story:hall]
+  ^-  (quip move _this)
+  ?>  ?=(%gram -.sto)
+  =*  messages  messages.str
+  =/  circle=circle:hall  [`@p`(slav %p &2:wir) &3:wir]
+  =/  nes=(unit (list envelope:hall))  (~(get by messages) circle)
```

Comment:
Probably put the body for these on its own line underneath, they're kinda long otherwise.

Reply:
Ping?